apps/google-analytics-4/lambda/src/services/googleApiService.ts

```diff
@@ -44,14 +45,51 @@ export class GoogleApiService {
     return accountSummaries;
   }
 
-  sortReportResponse = (rows: ReportRowType[]) => {
-    rows.filter(
-      (row: ReportRowType) => row.dimensionValues?.length && row.dimensionValues[0].value
+  formatDate = (dateValue: Date) => dateValue.toISOString().slice(0, 10).split('-').join('');
+
+  deriveDate = (dateValue: string) =>
+    new Date(
+      `${dateValue.substring(0, 4)}-${dateValue.substring(4, 6)}-${dateValue.substring(6, 8)}`
     );
 
-    rows.sort((v1: ReportRowType, v2: ReportRowType) => {
-      return Number(v1.dimensionValues[0].value) - Number(v2.dimensionValues[0].value);
+  mapDates = (startDate: string, endDate: string) => {
```

Comment:
nit: I don't think this name reflects what the logic is doing here. This is more building a prefix of dates keys. I think something like `prebuildDateKeyArray` would be more appropriate

Reply:
really like this idea, will do!